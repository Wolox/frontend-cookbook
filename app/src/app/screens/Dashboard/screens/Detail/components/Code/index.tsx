import React, { useState } from 'react';
import ReactMarkdown from 'react-markdown';

import CodeSnippet from '~components/CodeSnippet';
import Tree from '~components/Tree';
import { SelectedFile, TreeRecipe } from '~constants/interfaces/recipe';

import styles from './styles.module.scss';

interface Props {
  title: string;
  downloadCommand?: string;
  source: TreeRecipe;
  onDownload: () => void;
}

const defaultSelectedFile = {
  id: '',
  name: '',
  src: '',
  lang: '',
  isBinary: false
};

function Code({ title, source, downloadCommand, onDownload }: Props) {
  const [fileData, setFileData] = useState<SelectedFile>(defaultSelectedFile);
  const handleFileSelect = (data: SelectedFile = defaultSelectedFile) => {
    setFileData(data);
  };

  return (
    <div className={styles.filesStructureContainer}>
      <div className={`m-bottom-6 ${styles.header}`}>
        <div className="row">
          {downloadCommand && (
            <div className="column start m-left-6">
              <p className="subtitle m-bottom-2">Install using @wolox/cookbook CLI:</p>
              <div className="m-bottom-2">
                <pre className={styles.installInstructions}>{downloadCommand}</pre>
              </div>
            </div>
          )}
          <div className="column start m-left-6">
            <p className="subtitle m-bottom-2">Download files instead:</p>
            <button className={styles.downloadButton} type="button" onClick={onDownload}>
              Download code
            </button>
          </div>
        </div>
      </div>
      <h3 className={styles.fileName}>{fileData.name || '(Choose a file from the picker)'}</h3>
      <div className={styles.codeContainer}>
        <Tree handleSelect={handleFileSelect} source={source} title={title} activeId={fileData.id} />
        <div className={styles.codeSnippets}>
          {fileData.lang === 'md' ? (
            <ReactMarkdown
              className="full-width column markdown-container"
              source={fileData.src}
              escapeHtml={false}
            />
          ) : (
            <CodeSnippet
              className={`${styles.code} ${styles.codeVisible}`}
              code={fileData.isBinary ? 'Not supported file' : fileData.src}
              lang={fileData.lang}
            />
          )}
        </div>
      </div>
    </div>
  );
}

export default Code;
