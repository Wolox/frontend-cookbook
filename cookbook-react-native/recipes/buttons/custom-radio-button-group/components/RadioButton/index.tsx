import React, { Key } from 'react';
import { TouchableOpacity, ViewStyle, View, TextProps } from 'react-native';
import CustomText from '@textsRecipes/custom-text';

import styles from './styles';
import { getBackgroundColor, getBorderColor } from './utils';

interface Props<K, T> {
  option: T;
  optionKey: K;
  selected: boolean;
  disabled?: boolean;
  onPress: (optionKey: K) => void;
  textProps?: TextProps;
  style?: ViewStyle;
  radioButtonStyle?: ViewStyle;
}

function RadioButton<K extends Key, T>({
  selected = false,
  option,
  optionKey,
  disabled,
  onPress,
  textProps,
  style,
  radioButtonStyle
}: Props<K, T>) {
  const handlePress = () => onPress(optionKey);
  const isSelected = !disabled && selected;
  const backgroundColor = getBackgroundColor(isSelected);
  const borderColor = getBorderColor(disabled);
  return (
    <TouchableOpacity style={[styles.container, style]} onPress={handlePress} disabled={disabled}>
      <CustomText gray={disabled} textProps={textProps}>
        {option}
      </CustomText>
      <View style={[styles.circleContainer, { borderColor }, radioButtonStyle]}>
        {isSelected && <View style={[styles.circle, { backgroundColor }]} />}
      </View>
    </TouchableOpacity>
  );
}

export default RadioButton;
