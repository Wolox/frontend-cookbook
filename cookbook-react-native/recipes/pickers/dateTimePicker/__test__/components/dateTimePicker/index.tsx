import React from 'react';

import CustomDateTimePicker from '@pickersRecipes/dateTimePicker';

import renderer from 'react-test-renderer';

describe(' CustomDateTimePicker', () => {
  test('to match snapshot', () => {
    const component = renderer
      .create(<CustomDateTimePicker accessibilityLabel={''} is24hs={false} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
