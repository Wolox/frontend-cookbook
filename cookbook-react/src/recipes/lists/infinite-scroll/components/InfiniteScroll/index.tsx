import React, { useCallback, useEffect, useRef } from 'react';
import i18next from 'i18next';

import styles from './styles.module.scss';

type ElementType = JSX.Element | string;

interface Props {
  children: React.ReactNode;
  onLoadMore: () => void;
  isLoading: boolean;
  hasMore: boolean;
  loader?: ElementType;
  endMessage?: ElementType;
  className?: string;
}

const getAlertMessage = (defaultText: string, element?: ElementType, loading?: boolean) => {
  if (!element) {
    return (
      <div className={styles.alert}>
        {loading && <div className={styles.spinner} />} {defaultText}
      </div>
    );
  }

  if (typeof element === 'string') {
    return (
      <div className={styles.alert}>
        {loading && <div className={styles.spinner} />} {element}
      </div>
    );
  }

  return element;
};

function InfiniteScroll({ children, onLoadMore, isLoading, hasMore, loader, endMessage, className }: Props) {
  const contentListRef = useRef<HTMLDivElement>(null);
  const containerRef = useRef<HTMLDivElement>(null);

  const handleNextPage = useCallback(
    (entries: IntersectionObserverEntry[]) => {
      const [element] = entries;
      if (element.isIntersecting && contentListRef.current && hasMore && !isLoading) {
        onLoadMore();
      }
    },
    [hasMore, isLoading, onLoadMore]
  );

  useEffect(() => {
    const observer = new IntersectionObserver(handleNextPage, {
      threshold: 0.9
    });

    if (contentListRef.current) {
      observer.observe(contentListRef.current);
    }

    return () => observer.disconnect();
  }, [handleNextPage]);

  return (
    <>
      <div className={className} ref={containerRef}>
        {children}
      </div>
      {isLoading && getAlertMessage(i18next.t('InfiniteScroll:loading'), loader, true)}
      {!hasMore && getAlertMessage(i18next.t('InfiniteScroll:end'), endMessage)}
      <div ref={contentListRef} />
    </>
  );
}

export default InfiniteScroll;
