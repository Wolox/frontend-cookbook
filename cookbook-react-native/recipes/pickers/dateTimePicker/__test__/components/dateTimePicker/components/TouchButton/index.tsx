import React from 'react';
import renderer from 'react-test-renderer';
import { TouchButton } from '@pickersRecipes/dateTimePicker/components/TouchButton';

describe('TouchButton', () => {
  test('to match snapshot', () => {
    const component = renderer.create(<TouchButton />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
