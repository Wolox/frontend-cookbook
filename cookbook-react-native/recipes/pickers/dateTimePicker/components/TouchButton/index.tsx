import React from 'react';
import { TouchableOpacity, Image, Text } from 'react-native';
import i18next from 'i18next';
import { ITouchButton } from '@pickersRecipes/dateTimePicker/interfaces';

import '@pickersRecipes/dateTimePicker/i18n';
import styles from '@pickersRecipes/dateTimePicker/style';

export const TouchButton = ({
  styleBtn,
  showDatePicker,
  icon,
  styleIcon,
  styleTextBtn,
  textButton
}: ITouchButton) => (
  <TouchableOpacity
    style={[styles.defaultButtonStyle, styleBtn]}
    onPress={showDatePicker}>
    {icon ? (
      <Image
        source={icon}
        style={[styles.defaultIcon, styleIcon]}
        resizeMode="cover"
        resizeMethod="scale"
      />
    ) : (
      <Text style={[styles.defaultTextBtn, styleTextBtn]}>
        {textButton || i18next.t('CUSTOM_DATETIME_PICKER:OPEN')}
      </Text>
    )}
  </TouchableOpacity>
);
