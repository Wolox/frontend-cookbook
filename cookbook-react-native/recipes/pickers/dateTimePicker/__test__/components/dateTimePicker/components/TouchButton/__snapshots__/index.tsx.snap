// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TouchButton to match snapshot 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignContent": "center",
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "borderRadius": 20,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "padding": 14,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 16,
          "fontWeight": "bold",
        },
        undefined,
      ]
    }
  >
    CUSTOM_DATETIME_PICKER:OPEN
  </Text>
</View>
`;
